django/db/models/sql/where.py

```diff
@@ -129,12 +129,14 @@ def as_sql(self, compiler, connection):
             # Convert if the database doesn't support XOR:
             #   a XOR b XOR c XOR ...
             # to:
-            #   (a OR b OR c OR ...) AND (a + b + c + ...) == 1
+            #   (a OR b OR c OR ...) AND MOD(a + b + c + ..., 2) == 1
             lhs = self.__class__(self.children, OR)
             rhs_sum = reduce(
                 operator.add,
                 (Case(When(c, then=1), default=0) for c in self.children),
             )
+            if len(self.children) > 2:
+                rhs_sum = Func(rhs_sum, 2, function="MOD")
             rhs = Exact(1, rhs_sum)
```

Comment:
I wonder if we need the new `if` guard. In terms of correctness, the following works for any length of children:

```suggestion
            rhs = Exact(1, Func(rhs_sum, 2, function="MOD"))
```

We should check if there are any performance penalties to justify the new guard.